packages/shared/src/components/notifications/Toast.tsx

```diff
@@ -41,17 +41,18 @@ const Toast = ({
     () => client.getQueryData(TOAST_NOTIF_KEY),
     {
       enabled: false,
-      onSuccess: async (data) => {
-        if (!data) {
-          return;
-        }
-
-        await nextTick(); // wait (1ms) for the component to render so animation can be seen
-        startAnimation(data.timer);
-      },
     },
   );
 
+  if (!toastRef.current && toast?.message) {
+    toastRef.current = toast;
+    startAnimation(toast.timer);
```

Comment:
Since `startAnimation` is calling `setTimer` this effectively sets state while rendering which is bad an can lead to infinite loops. I get it that you guard with refs but there must be a better solution.

Won't block if you tested it and you decided with others to move CSS solution to backlog but we should be very careful with this.

Reply:
Having spoke to Chris, have confirmed we are okay to continue especially with the other ticket now on the backlog for the refactor. but I will keep an eye on this incase if does have a negative effect on anything